### ISSUE TITLE
Failed to reproduce BigGait on CCPG

### ISSUE THREAD

**Issue by Graysonggg** [ShiqiYu/OpenGait#248]
Hello again, Biggait is impressive, so I tried to reproduce it but encountered a problem. I used the checkpoint in the link https://drive.google.com/drive/folders/1zrWPUsrbCpwxoLgfom3d2irgxkBqtXqc to execute the test command：
`CUDA_VISIBLE_DEVICES=0,1 python -m torch.distributed.launch --nproc_per_node=2 opengait/main.py --cfgs ./configs/biggait/BigGait_CCPG.yaml --phase test`
then found that I could not reproduce the results on CCPG in the paper:

![image](https://github.com/user-attachments/assets/95bb7af2-4340-4be0-959b-764f0fc0b042)

I only modified the evaluation batch size (number of GPUs) as 2 in cfg file, and did not make any other changes. I would like to know where the problem lies.

In terms of data preprocessing, I used some code from datasets/CCPG/organize_ccpg.py to generate only the rgb and ratio pkl files for ccpg, and have verified that what I generated is not silhouettes.：

```
import os
import pickle
import numpy as np
import cv2
from tqdm import tqdm
import argparse

if __name__ == '__main__':
    parser = argparse.ArgumentParser(description='CCPG dataset Preprocessing.')
    parser.add_argument('--rgb_path', default='', type=str,
                        help='Root path of raw RGB dataset.')
    parser.add_argument('-o', '--output_path', default='',
                        type=str, help='Output path of pickled dataset.')
    args = parser.parse_args()
    RGB_SIZE = (128, 128)
    for _id in tqdm(sorted(os.listdir(args.rgb_path))):
        for _type in sorted(os.listdir(os.path.join(args.rgb_path, _id))):
            for _view in sorted(os.listdir(os.path.join(args.rgb_path, _id, _type))):
                imgs = []
                ratios = []
                for img_file in sorted(os.listdir(os.path.join(args.rgb_path, _id, _type, _view))):
                    img_path = os.path.join(
                        args.rgb_path, _id, _type, _view, img_file)
                    if not os.path.exists(img_path):
                        print("Not Found: "+img_path)
                        continue
                    img = cv2.imread(img_path)
                    ratio = img.shape[1]/img.shape[0]
                    img = np.transpose(cv2.cvtColor(cv2.resize(
                        img, RGB_SIZE), cv2.COLOR_BGR2RGB), (2, 0, 1))
                    imgs.append(img)
                    ratios.append(ratio)
                if len(imgs) > 0:
                    output_path = os.path.join(
                        args.output_path, _id, _type, _view)
                    os.makedirs(output_path, exist_ok=True)
                    pickle.dump(np.asarray(imgs), open(os.path.join(
                        output_path, _view+"-rgbs.pkl"), "wb"))
                    pickle.dump(np.asarray(ratios), open(os.path.join(
                        output_path, _view+"-ratios.pkl"), "wb"))
                else:
                    print("No imgs Found: " +
                          os.path.join(args.rgb_path, _id, _type, _view))
                    continue

```

My preliminary inference is that the reason may be that datasets/CCPG/organize_pg.py has not been updated. If that's the case, can you update the code related to biggait that can be reproduced successfully? Thank you very much indeed!

**Comment by Bugjudger** [comment]
Thanks for your interest. Pls use [code](https://github.com/ShiqiYu/OpenGait/tree/master/datasets/pretreatment_rgb.py) for preprocessing. BigGait rely on truthful human structural characteristics for gait recognition. So the above preprocessing code will add black edges for various images with different size to keep body ratio.

**Comment by Graysonggg** [comment]
I reproduce successfully using your mentioned pretreatment script, thank you very much.

**Comment by Lxzzzzzzzzzzz** [comment]
你好，我想知道，./pretrained_LVMs/dinov2_vits14_pretrain.pth这个文件在哪里下载呀

**Comment by Bugjudger** [comment]
> 你好，我想知道，./pretrained_LVMs/dinov2_vits14_pretrain.pth这个文件在哪里下载呀

https://github.com/facebookresearch/dinov2?tab=readme-ov-file#pretrained-models

**Comment by Bugjudger** [comment]
> hello， I want to train a pretrained_mask_branch: ./pretrained_LVMs/MaskBranch_vits14.pt. What should I do

You can directly train BigGait without providing pretrained_mask_branch. Models will auto-train this simple Mask Branch in first 500 iterations. The training of Mask Branch is unstable, it depends on the divisibility of LVM features. Our pretrain Mask Branch is recommended. Try to use different random seed and drop out rate for your own training.
